File: 0589-n-ary-tree-preorder-traversal/0589-n-ary-tree-preorder-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<int> preorder(Node* root) {
        if(root == NULL){
            return {};
        }
        vector<int> ans;
        ans.push_back(root->val);
        for(int i = 0; i < root->children.size(); i++){
            vector<int> tmp = preorder(root->children[i]);
            ans.insert(ans.end(), tmp.begin(), tmp.end());
        }
        return ans;
    }
};

File: 0589-n-ary-tree-preorder-traversal/0589-n-ary-tree-preorder-traversal.cs
/*
// Definition for a Node.
public class Node {
    public int val;
    public IList<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val,IList<Node> _children) {
        val = _val;
        children = _children;
    }
}
*/

public class Solution {
    public IList<int> Preorder(Node root) {
        List<int> list = new List<int>();
        if(root == null){
            return list;
        }
        list.Add(root.val);
        for(int i = 0; i < root.children.Count; i++){
            list.AddRange(Preorder(root.children[i]));
        }
        return list;
    }
}